### ISSUE TITLE
OMEMO doesn't work

### ISSUE THREAD

**Issue by roughnecks** [conversejs/roundcube-plugin#22]
Hello,
I see this config: 'omemo_default' => true,
but OMEMO just doesn't work and I can't even see the "keys" tab when clicking on my profile.

For all OMEMO messages I get an info message about my client not supporting it.

Am I missing something?
Thanks

**Comment by drlight17** [comment]
Hi,
I think it depends on the CDN version you use and it is not the question of this plugin. You should ask in the proper conversejs repo. Maybe [here](https://github.com/conversejs/converse.js/issues?q=is%3Aissue+is%3Aopen)?

**Comment by roughnecks** [comment]
Why? I use `$rcmail_config['converse_cdn'] = 'https://cdn.conversejs.org/10.1.2';`
as per default config file

And OMEMO works just fine in `mod_conversejs` in Prosody.

**Comment by roughnecks** [comment]
libsignal isn't loaded:

![image](https://github.com/conversejs/roundcube-plugin/assets/1666536/ef90afcf-1c2d-41ac-a0dd-4b236822e978)

https://m.conversejs.org/docs/html/setup_dev_environment.html#dependency-libsignal

**Comment by bronkoo** [comment]
> Am I missing something?
> Thanks

You have to uncomment include_script in `converse.php`:

`// TODO  add omemo encryption support (no messages received, error for now)`
`$this->include_script('https://cdn.conversejs.org/3rdparty/libsignal-protocol.min.js');`

Now its working for me.

**Comment by roughnecks** [comment]
This plugin is unusable for me, I gave up
If I disconnect and then connect a second time, more than half my chats and contacts aren't there in list
Also I tried your fix but when clicking on the lock icon it said the other person wasn't using a client capable of OMEMO - and I talk with that person on a regular basis on my standard conversejs client

Thanks tho.

**Comment by bronkoo** [comment]
yes, only a short history, missing contacts, file transfer doesn't work and OMEMO starts after cookie deleting in browser...

**Comment by bronkoo** [comment]
> Also I tried your fix but when clicking on the lock icon it said the other person wasn't using a client capable of OMEMO

When I enabled mod_conversejs (ejabberd) on server site too, I got these... so don't configure conversejs module on xmpp server, only websocket or bosh for your roundcube conversejs plugin connetion.
